packages/mobile/src/utils/accountChecker.test.ts

```diff
@@ -0,0 +1,63 @@
+import * as Keychain from 'react-native-keychain'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import { deleteNodeData } from 'src/geth/geth'
+import { resetStateOnInvalidStoredAccount } from 'src/utils/accountChecker'
+import { walletAddressSelector } from 'src/web3/selectors'
+import { getMockStoreData } from 'test/utils'
+import { mocked } from 'ts-jest/utils'
+
+jest.mock('src/geth/geth')
+
+const mockedKeychain = mocked(Keychain)
+
+beforeEach(() => {
+  jest.clearAllMocks()
+})
+
+describe('resetStateOnInvalidStoredAccount', () => {
+  it("returns the same state when there's no account", async () => {
+    const state = getMockStoreData({ web3: { account: null } })
+    expect(walletAddressSelector(state)).toEqual(null)
+    const result = await resetStateOnInvalidStoredAccount(state)
+
+    expect(result === state).toEqual(true)
```

Comment:
this is the first time i've seen this 😄  it clearly works but is the more conventional form `expect(result).toEqual(state)`?

Reply:
Yes I wanted to avoid ambiguity for readers 😄 
I initially wrote it as `expect(result).toBe(state)` which uses `Object.is` under the hood.
But I really wanted identity comparison. `toEqual` doesn't fit either.
https://stackoverflow.com/a/63003041/158525

Reply:
ah, i see what you're trying to do now! makes sense